### ISSUE TITLE
House_Price_Prediction

### ISSUE THREAD

**Issue by amrutadpatil2001-droid** [amrutadpatil2001-droid/Python_Projects#12]
import pandas as pd
df = pd.read_csv("house_price_data.csv")
print(df.head())
print(df.info())
print(df.isnull().sum())

#Visualization
import matplotlib.pyplot as plt
import seaborn as sns
sns.pairplot(df)
plt.show()

#Handle Missing Values
df = df.dropna()

#Convert Categorical
df = pd.get_dummies(df, columns=['location'], drop_first=True)

#Define features & target
X = df.drop('totalprice', axis=1)
y = df['totalprice']

#Train test split
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

#Model using scikit-learn
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(X_train, y_train)
y_pred = model.predict(X_test)

#Evaluate Model
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
mae = mean_absolute_error(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)
print("MAE:", mae)
print("MSE:", mse)
print("R²:", r2)

#Visualization
plt.scatter(y_test, y_pred)
plt.xlabel("Actual Prices")
plt.ylabel("Predicted Prices")
plt.title("Actual vs Predicted")
plt.show()

#Improve Model
# Polynomial Feature
from sklearn.preprocessing import PolynomialFeatures
poly = PolynomialFeatures(degree=2)
X_train_poly = poly.fit_transform(X_train)
X_test_poly = poly.transform(X_test)
model.fit(X_train_poly, y_train)
y_pred_poly = model.predict(X_test_poly)

# Decision Tree
from sklearn.tree import DecisionTreeRegressor
dt = DecisionTreeRegressor()
dt.fit(X_train, y_train)

# Random Forest
from sklearn.ensemble import RandomForestRegressor
rf = RandomForestRegressor(n_estimators=100)
rf.fit(X_train, y_train)

importance = rf.feature_importances_
for i, col in enumerate(X.columns):
    print(col, importance[i])
